CMake/RealmCore.cmake

```diff
@@ -109,6 +113,14 @@ function(download_realm_core core_version)
     set_property(TARGET realm PROPERTY IMPORTED_LOCATION_RELEASE ${core_library_release})
     set_property(TARGET realm PROPERTY IMPORTED_LOCATION ${core_library_release})
 
+    if (LINUX)
+        find_package(Threads)
+        find_package(PkgConfig REQUIRED)
+        pkg_search_module(OPENSSL REQUIRED openssl)
+        message(STATUS "Using OpenSSL ${OPENSSL_VERSION}")
+        set_property(TARGET realm PROPERTY INTERFACE_LINK_LIBRARIES ${CMAKE_THREAD_LIBS_INIT} ${OPENSSL_LIBRARIES})
```

Comment:
This also needs to be done in `define_built_realm_core_target` so that the correct libraries are linked in when building core and sync from source.